TriangleApp/app/src/main/java/triangle/triangleapp/helpers/MediaHelper.java

```diff
@@ -0,0 +1,192 @@
+
+                            int read = fis.read(bytes, 0, size);
+                            if (read < size) {
+                                int remain = size - read;
+                                while (remain > 0) {
+                                    read = fis.read(tmpBuff, 0, remain);
+                                    System.arraycopy(tmpBuff, 0, bytes, size - remain, read);
+                                    remain -= read;
+                                }
+                            }
+                        } catch (IOException e) {
+                            Log.e(TAG, "IoExc", e);
+                        }
+                        webSocket.sendStream(bytes);
+                        file.delete();
+                    }
+                }
+            }
+        });
+
+        mMediaRecorder.setOnErrorListener(new MediaRecorder.OnErrorListener() {
+            @Override
+            public void onError(MediaRecorder mediaRecorder, int what, int extra) {
+                Log.d(TAG, "Error in mediarecorder, " + what + ", " + extra);
+            }
+        });
+
+        mMediaRecorder.setMaxDuration(mediaRecorderMaxDuration);
+    }
+
+    private File getOutputMediaFile(int type) {
+        File mediaStorageDir = new File(Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS), "MyCameraApp");
+        if (!mediaStorageDir.exists()) {
+            if (!mediaStorageDir.mkdirs()) {
+                Log.d("MyCameraApp", "failed to create directory");
+                return null;
+            }
+        }
+
+        String timeStamp = new SimpleDateFormat("yyyyMMdd_HHmmss").format(new Date());
+        File mediaFile = null;
+        switch (type) {
+            case MEDIA_TYPE_IMAGE:
+                mediaFile = new File(mediaStorageDir.getPath() + File.separator + "IMG_" + timeStamp + ".jpg");
+                break;
+            case MEDIA_TYPE_VIDEO:
+                mediaFile = new File(mediaStorageDir.getPath() + File.separator + "VID_" + timeStamp + ".mp4");
+                break;
+        }
+        return mediaFile;
+    }
+
+    private void startStreaming(boolean firstStart) {
+        initializeVideoRecorder(firstStart);
+
+        if (!prepareVideoRecorder()) {
+            releaseMediaRecorder();
+            return;
+        }
+
+        try {
+            mMediaRecorder.start();
+        } catch (Exception ex) {
+            Log.e(TAG, "Error during start", ex);
+        }
+        isRecording = true;
+    }
+
+    private void stopStreaming(boolean fullStop) {
+        // stop recording and release camera
+        mMediaRecorder.stop();  // stop the recording
+        mMediaRecorder.reset();
+        if (fullStop) {
+            releaseMediaRecorder(); // release the MediaRecorder object
+        }
+        mCamera.lock();         // take camera access back from MediaRecorder
+    }
+
+    private boolean prepareVideoRecorder() {
```

Comment:
Added comment to method: prepareVideoRecorder.
/**
 * Prepare mediarecorder to record.
 * @return
 */